test/spark.js

```diff
@@ -363,3 +365,102 @@ test('calculateDelayBeforeNextTask() handles one task per round', () => {
   })
   assertEquals(delay, 60_000)
 })
+
+const mockPeerIdResponse = {
+  peerID: '12D3KooWGQmdpbssrYHWFTwwbKmKL3i54EJC9j7RRNb47U9jUv1U',
+  signature: '0x1234567890abcdef'
+}
+
+const mockEmptyPeerIdResponse = {
+  peerID: '',
+  signature: '0x'
+}
+
+// Mock contract factory
+function createMockContract (mockResponses) {
+  return {
+    getPeerData: async (minerId) => {
+      const response = mockResponses[minerId]
+      if (!response) {
+        throw new Error(`Miner ID ${minerId} not found in contract`)
+      }
+      return response
+    }
+  }
+}
+
+test('smart-contract-client: getMinerPeerIdFromSmartContract returns peer ID for valid miner ID', async () => {
+  // Create mock contract with predefined responses
+  const peerId = 12345
+  const mockContract = createMockContract({
+    [peerId]: mockPeerIdResponse
+  })
+
+  const actualPeerId = await getMinerPeerIdFromSmartContract(`f0${peerId}`, {
+    getSmartContractClientFn: () => mockContract
+  })
+
+  assertEquals(actualPeerId, mockPeerIdResponse.peerID)
+})
+
+test('miner-info: getMinerPeerId integration test with real miner f01234', async () => {
+  const peerId = await getMinerPeerIdFromSmartContract('f012345', RPC_URL)
+  assertEquals(typeof peerId, 'string', 'Expected peerId to be a string')
+  assert(peerId.length > 0, 'Expected peerId to be non-empty')
+  assertEquals(peerId, '12D3KooWGQmdpbssrYHWFTwwbKmKL3i54EJC9j7RRNb47U9jUv1U')
```

Comment:
The first assertion is redundant, I believe the second assertion already covers that check.

```suggestion
  assertEquals(peerId, '12D3KooWGQmdpbssrYHWFTwwbKmKL3i54EJC9j7RRNb47U9jUv1U')
```

Reply:
Addressed in 3a60176